tools/platformio-build.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2014-present PlatformIO <contact@platformio.org>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Arduino
+
+Arduino Wiring-based Framework allows writing cross-platform software to
+control devices attached to a wide range of Arduino boards to create all
+kinds of creative coding, interactive objects, spaces or physical experiences.
+
+http://arduino.cc/en/Reference/HomePage
+"""
+
+# Extends: https://github.com/platformio/platform-espressif8266/blob/develop/builder/main.py
+
+from os.path import isdir, join
+
+from SCons.Script import DefaultEnvironment
+
+env = DefaultEnvironment()
+platform = env.PioPlatform()
+
+FRAMEWORK_DIR = platform.get_package_dir("framework-arduinoespressif8266")
+assert isdir(FRAMEWORK_DIR)
+
+
+env.Prepend(
+    CPPDEFINES=[
+        ("ARDUINO", 10600),
+        "LWIP_OPEN_SRC"
+    ],
+    CPPPATH=[
+        join(FRAMEWORK_DIR, "tools", "sdk", "include"),
+        join(FRAMEWORK_DIR, "tools", "sdk", "lwip", "include"),
+        join(FRAMEWORK_DIR, "tools", "sdk", "libc",
+             "xtensa-lx106-elf", "include"),
+        join(FRAMEWORK_DIR, "cores", env.BoardConfig().get("build.core"))
+    ],
+    LIBPATH=[
+        join(FRAMEWORK_DIR, "tools", "sdk", "lib"),
+        join(FRAMEWORK_DIR, "tools", "sdk", "ld"),
+        join(FRAMEWORK_DIR, "tools", "sdk", "libc", "xtensa-lx106-elf", "lib")
+    ],
+    LIBS=[
+        "mesh", "wpa2", "smartconfig", "espnow", "pp", "main", "wpa", "lwip_gcc",
+        "net80211", "wps", "crypto", "phy", "hal", "axtls", "gcc",
+        "m", "c", "stdc++"
+    ]
+)
+
+env.Append(
+    LIBSOURCE_DIRS=[
+        join(FRAMEWORK_DIR, "libraries")
+    ],
+    LINKFLAGS=[
+        "-Wl,-wrap,system_restart_local",
+        "-Wl,-wrap,spi_flash_read"
```

Comment:
Arduino Build System is a self-written bunch of files where they play with common problems which were resolved by `make` and other build tools a lot of years ago. We don't depend on Arduino IDE, their build system or even Java. For us, Arduino is just a `framework` with some "build logic".

We didn't invent the new wheel and our PIO Build System is based on [SCons](http://scons.org) construction tool. It's 100% pure Python implementation with Zero-dependencies to other packages or even to the OS software. We applied a lot of patches to SCons to make it better for embedded using and now it's super-fast and reliable.

In `platform.txt` all flags are located in ONE LINE, where in PlatformIO we separate them into [multiple scopes](http://docs.platformio.org/en/latest/projectconf.html#build-flags). Later, a developer is able to "unflag" some of them directly from project configuration file or add new. Here is very important that each flag/option is located in correct scope. 

We would be thankful if you support this build script. I see that it doesn't require a lot of efforts. Take a look at [platformio-build.py for ESP32](https://github.com/espressif/arduino-esp32/commits/master/tools/platformio-build.py) by @me-no-dev . I don't see here a hundred commits.

In any case, you can ping me or @valeros any time and we will help as soon as possible.